### PR TITLE
OMI_physics_body

### PR DESCRIPTION
**Pull request by aaronfranke**
This PR adds OMI_physics_body. It supersedes #75 from earlier this year. This PR takes #75, overhauls the text to be more readable, updates the table of equivalent types in major engines, adds some discussed features, fixes and completes* the schema, and adds a folder of example GLTF files.

Preview: https://github.com/aaronfranke/gltf-extensions/tree/OMI_physics_body/extensions/2.0/OMI_physics_body

I have the document listed as a Stage 1 proposal. This PR meets all of the requirements of [Stage 1](https://github.com/omigroup/gltf-extensions#stage-1) and also meets nearly all of the requirements of [Stage 2](https://github.com/omigroup/gltf-extensions#stage-2), except that for all practical purposes, since this depends on OMI_collider, the spec can't be truly complete without OMI_collider merged in first, so this is waiting on #127.

Stage 1

- [x] High level outline published as a GitHub PR in this repository. (this PR)
- [x] [Signed W3C CLA](https://www.w3.org/community/about/process/cla/). (yes)

Stage 2

- [x] Previous requirements from Stage 1. (yes)
- [x] One metaverse implementation of the extension in development. ([yes, in Godot Engine](https://github.com/godotengine/godot/pull/69266))
- [x] Valid and publicly accessible sample assets. (yes, see `examples/` subfolder, but note these also use OMI_collider)
- [ ] Valid JSON schema describing the extension in its entirety. (this PR has a schema for OMI_physics_body, but a full description in its entirety depends on OMI_collider)
- [x] Markdown document explaining the purpose, features, and properties of the extension. (yes, see the `OMI_physics_body/README.md` file in this PR)

Stage 3

- [ ] Previous requirements from Stage 2. (not yet)
- [ ] Two or more metaverse implementations. (not yet, just Godot Engine)
- [x] glTF Validator implementation. ([yes, branch here](https://github.com/aaronfranke/glTF-Validator/tree/omi_physics_body))
- [x] Publicly available content creation tool for generating assets for this extension. (the Godot Engine implementation can export OMI_physics_body)
- [ ] All implementations, sample assets, and content creation tools should consume or produce assets that pass validation. (awaiting more implementations)

### CODE REVIEW

**Comment by lyuma** [comment]
feedback on physics properties:
isKinematic - we call it type.
mass - useful
inertiaTensor - useful. Moment of Intertia is very important for defining an object's rotation. Not all engines might support a full Vector3 tensor. See here for definition: https://en.wikipedia.org/wiki/Moment_of_inertia#Inertia_tensor
centerOfMass - no. it should be required to offset the physics_body node such that it exists at its center of mass
linearVelocity - useful. my feedback: one issue is linear and angular velocity would need to be held in a script to be assigned at runtime, since many engines don't support setting this statically in a scene.
angularVelocity - same as linearVelocity

**Comment by lyuma** [comment]
should OMI_physics_joint be a separate extension? is the behavior of doors falling off when the extension is unimplemented acceptable?
https://github.com/eoineoineoin/glTF_Physics#joints

MSFT defines a joint with constraints on XYZ axes. But because MSFT joint must be defined on a node (same as Unity), you cannot offset the joint axes with respect to the physics body. In Godot's system Joint is its own node, so it could have its own arbitrary axis of rotation.

We should see if the joint spec is sufficient but MSFT claims it can be used to build hinges and other types of joints.

File: extensions/2.0/OMI_physics_body/README.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+If an `OMI_collider`'s `"isTrigger"` setting does not match the body it's a part of, implementations should ensure the per-collider setting is preserved.
+
+### Mass
+
+The `"mass"` property is a number that defines how much mass this physics body has in kilograms. Not all body types can make use of mass, such as triggers or non-moving bodies, in which case the mass can be ignored. If not specified, the default value is 1 kilogram.
+
+### Linear Velocity
+
+The `"linearVelocity"` property is an array of three numberes that defines how much linear velocity this physics body starts with in meters per second. Not all body types can make use of linear velocity, such as non-moving bodies, in which case the linear velocity can be ignored. If not specified, the default value is zero.
```
Comment (fire) [inline extensions/2.0/OMI_physics_body/README.md:151]:
```suggestion
The `"linearVelocity"` property is an array of three numbers that defines how much linear velocity this physics body starts with in meters per second. Not all body types can make use of linear velocity, such as non-moving bodies, in which case the linear velocity can be ignored. If not specified, the default value is zero.
```

File: extensions/2.0/OMI_physics_body/README.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+### Angular Velocity
+
+The `"angularVelocity"` property is an array of three numberes that defines how much angular velocity this physics body starts with in radians per second. Not all body types can make use of angular velocity, such as non-moving bodies, in which case the angular velocity can be ignored. If not specified, the default value is zero.
```
Comment (fire) [inline extensions/2.0/OMI_physics_body/README.md:155]:
```suggestion
The `"angularVelocity"` property is an array of three numbers that defines how much angular velocity this physics body starts with in radians per second. Not all body types can make use of angular velocity, such as non-moving bodies, in which case the angular velocity can be ignored. If not specified, the default value is zero.
```

**Review by fire — COMMENTED** [review]
Small corrections.

File: extensions/2.0/OMI_physics_body/README.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+## Known Implementations
+
+Godot Engine: https://github.com/godotengine/godot/pull/69266
+
+## Resources:
+
+Unity colliders: https://docs.unity3d.com/Manual/CollidersOverview.html
+Unreal Engine Physics: https://docs.unrealengine.com/4.27/en-US/InteractiveExperiences/Physics/Collision/Overview/
+Godot Physics Body: https://docs.godotengine.org/en/stable/classes/class_physicsbody.html
+Godot Area: https://docs.godotengine.org/en/stable/classes/class_area.html
```
Comment (fire) [inline extensions/2.0/OMI_physics_body/README.md:170]: Use markdown syntax for a list?

**Comment by robertlong** [comment]
Vote to merge this PR as the working draft succeeded. Screenshot 2023-01-05 at 3 21 55 PM

**Review by robertlong — APPROVED** [review]
Looks good 👍

**Pull request merged**